hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DataStreamer.java

```diff
@@ -934,14 +936,14 @@ void waitForAckedSeqno(long seqno) throws IOException {
             }
             try {
               dataQueue.wait(1000); // when we receive an ack, we notify on
-              long duration = Time.monotonicNow() - begin;
-              if (duration > writeTimeout) {
+              long duration = Time.monotonicNowNanos() - begin;
```

Comment:
`long duration = TimeUnit.NANOSECONDS.toMillis(Time.monotonicNowNanos() - begin);`
here how about it?

Reply:
@haiyang1987 ,Sir. Thanks for your reviewing.  But the approach you proposed still has problem.
for example,  suppose that `Time.monotonicNowNanos() - begin = 56000ns`.   TimeUnit.NANOSECONDS.toMillis(56000) will return 0. So, the duration is 0.

Reply:
Get it , thanks @hfutatzhanghb comment.

```
if (TimeUnit.NANOSECONDS.toMillis(duration) > writeTimeout) {
    LOG.error("No ack received, took {}ms (threshold={}ms). "
        + "File being written: {}, block: {}, "
        + "Write pipeline datanodes: {}.",
        TimeUnit.NANOSECONDS.toMillis(duration), writeTimeout, src, block, nodes);
    throw new InterruptedIOException("No ack received after " +
      TimeUnit.NANOSECONDS.toSeconds(duration) + "s and a timeout of " +
        TimeUnit.MILLISECONDS.toSeconds(writeTimeout) + "s");
}
```
how about it?

Reply:
> Get it , thanks @hfutatzhanghb comment.
> 
> ```
> if (TimeUnit.NANOSECONDS.toMillis(duration) > writeTimeout) {
>     LOG.error("No ack received, took {}ms (threshold={}ms). "
>         + "File being written: {}, block: {}, "
>         + "Write pipeline datanodes: {}.",
>         TimeUnit.NANOSECONDS.toMillis(duration), writeTimeout, src, block, nodes);
>     throw new InterruptedIOException("No ack received after " +
>       TimeUnit.NANOSECONDS.toSeconds(duration) + "s and a timeout of " +
>         TimeUnit.MILLISECONDS.toSeconds(writeTimeout) + "s");
> }
> ```
> 
> how about it?

@hfutatzhanghb Sir, thanks a lot for your reviewing, you are welcome.  This part i did not use nanosecond as unit because i think printing error logs using nanosecond is not very intuitive, So I keep it the way it was.  As this PR's title described, its goal is for better debugging. Of course，it can be nanosecond, we can here other peoples' thoughts. Thanks again.

Reply:
@haiyang1987 Sir, Thanks a lot for your reviewing, you are welcome. Very nice suggestions here, Have updated it.